### ISSUE TITLE
mlpModule.final_fit function is not working

### ISSUE THREAD

**Issue by limengfromchina** [keras-team/autokeras#574]
### Reproducing Steps
  * Step 1: ...
  * Step 2: ...

### Setup Details
Include the details about the versions of:
 - OS type and version:
 - Python: 
 - autokeras: 
 - scikit-learn:
 - numpy:
 - keras:
 - scipy:
 - tensorflow:
 - pytorch:

**Comment by limengfromchina** [comment]
here is my code:

import numpy as np
from autokeras import MlpModule
from autokeras.nn.loss_function import classification_loss
from autokeras.nn.metric import Accuracy
from autokeras.preprocessor import OneHotEncoder, DataTransformerMlp

def transform_y(y_train):
    # Transform y_train.
    y_encoder = OneHotEncoder()
    y_encoder.fit(y_train)
    y_train = y_encoder.transform(y_train)
    return y_train, y_encoder

y_train, y_encoder = transform_y(Y_train)
data_transformer = DataTransformerMlp(X_train)
train_data = data_transformer.transform_train(X_train, Y_train)
test_data = data_transformer.transform_test(X_val, Y_val)

mlpModule = MlpModule(loss=classification_loss, metric=Accuracy, searcher_args={}, verbose=True)
fit_args = {
        "n_output_node": y_encoder.n_classes,
        "input_shape": X_train.shape,
        "train_data": train_data,
        "test_data": test_data}

mlpModule.fit(n_output_node=fit_args.get("n_output_node"),
                  input_shape=fit_args.get("input_shape"),
                  train_data=fit_args.get("train_data"),
                  test_data=fit_args.get("test_data"),
                  time_limit= 10 * 60)
mlpModule.final_fit(train_data, test_data, trainer_args=None, retrain=False)

**Comment by limengfromchina** [comment]
After i run this code I got error like this:
Exception occurred at train() : train_model() argument after ** must be a mapping, not NoneType

**Comment by limengfromchina** [comment]
after I  change the parameter retrain, I got similar error: Exception occurred at train() : 'NoneType' object has no attribute 'produce_model'.  who can help me with this issue, I'm really feeling desperate now...
